internal/controller/hubble/hubble.go

```diff
@@ -21,10 +21,16 @@ import (
 )
 
 const (
-	ciliumHubbleRelayExpectedServerName = "ui.hubble-relay.cilium.io"
-	ciliumHubbleRelayServiceName        = "hubble-relay"
+	ciliumHubbleRelayExpectedServerName     = "ui.hubble-relay.cilium.io"
+	gkeManagedHubbleRelayExpectedServerName = "hubble-relay.gke-managed-dpv2-observability.svc.cluster.local"
```

Comment:
It doesn't make sense to hardcode this, since you made the package name configurable, and the Service is somewhat discovered dynamically.
This service FQDN should be constructed as `<service name>.<package name>.svc.cluster.local`.
Cf. https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#a-aaaa-records

That is normal. The `ui.hubble-relay.cilium.io` domain name is an anomaly that we have to deal with.